Ocpbugs 25129: metallb should make sure that secondary nics are forwarding traffic (#83087)
openshift/openshift-docs: modules/nw-metallb-configure-secondary-interface.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-metallb-configure-secondary-interface_{context}"]

> = Configuring MetalLb with secondary networks

Reviewer: ```suggestion
= Configuring MetalLB with secondary networks
```

From {product-title} 4.14 the default behavior is to not allow forwarding of IP packets between network interfaces. Therefore when MetalLb is configured on a secondary interfaces which is common practice you need to add a machine configuration to enable IP forwarding for only the required interfaces. 
[NOTE]
====

> {product-title} clusters upgraded from 4.13 are not affected as a global parameter is set during upgrade to enable global IP forwarding.

Reviewer: ```suggestion
{product-title} clusters upgraded from 4.13 are not affected because a global parameter is set during upgrade to enable global IP forwarding.
```

====

> Follow this procedure to add a machine configuration to enable forwarding for the secondary interface for example `bridge-net` that MetalLb is configured on. 

Reviewer: ```suggestion
Follow this procedure to add a machine configuration to enable forwarding for the secondary interface, for example, `bridge-net` that MetalLB is configured on. 
```

.Procedure

> . Create the `MachineConfig` CR that enables IP forwarding for the specified secondary interface named `bridge-net`.

Reviewer: Up to you, but still suggesting it
```suggestion
. Create the `MachineConfig` CR to enable IP forwarding for the specified secondary interface named `bridge-net`.
```

.. Save the following YAML in the `enable-ip-forward.yaml` file:
+
[source,yaml]
----
apiVersion: machineconfiguration.openshift.io/v1
kind: MachineConfig
metadata:
  labels:
    machineconfiguration.openshift.io/role: <node_role> <1>
  name: 81-enable-global-forwarding
spec:
  config:
    ignition:
      version: 3.2.0
    storage:
      files:
      - contents:
          source: data:text/plain;charset=utf-8;base64,`echo -e "net.ipv4.conf.bridge-net.forwarding = 1\nnet.ipv6.conf.bridge-net.forwarding = 1\nnet.ipv4.conf.bridge-net.rp_filter = 0\nnet.ipv6.conf.bridge-net.rp_filter = 0" | base64 -w0`
           verification: {}
        filesystem: root
        mode: 644
        path: /etc/sysctl.d/enable-global-forwarding.conf
  osImageURL: ""
----
+

> <1> Node role where you want to enable IP forwarding for example `worker`

Reviewer: ```suggestion
<1> Node role where you want to enable IP forwarding, for example, `worker`
```

Reviewer 2: @kquinn1204 There is another way to enable forwarding by using flag ipForwarding
<pre>
oc patch network.operator cluster -p '{"spec":{"defaultNetwork":{"ovnKubernetesConfig":{"gatewayConfig":{"ipForwarding": "Global"}}}}}
</pre>

Author: Thank you @asood-rh I saw that "Change install-config manifests prior to deployment to reflect the new parameter and globally enable forwarding between *all* interfaces." Do I need to change install-config as I have also been assigned this https://issues.redhat.com/browse/TELCODOCS-1853 with mentions simply setting `ipForwarding: Global`. Seems to be a couple tied to this change. What is preferred method globally or as in this example ob specific secondary interface using MachineConfig.

:_mod-docs-content-type: PROCEDURE
[id="nw-metallb-configure-secondary-interface_{context}"]
= Configuring MetalLB with secondary networks

From {product-title} 4.14 the default network behavior is to not allow forwarding of IP packets between network interfaces. Therefore, when MetalLB is configured on a secondary interface, you need to add a machine configuration to enable IP forwarding for only the required interfaces. 

> [NOTE]

Reviewer 3: ```suggestion

[NOTE]
```
Renders OK on the docs.openshift.com preview, but you'll want to check the docs.redhat.com preview after merge; hard to say if the lack of the line space breaks anything there.
